bioptim/gui/plot.py

```diff
@@ -370,9 +372,14 @@ def __create_plots(self):
                             y_max = max([nlp.plot[variable].bounds.max.evaluate_at(j)[k] for j in range(nlp.ns)])
                         y_range, _ = self.__compute_ylim(y_min, y_max, 1.25)
                         ax.set_ylim(y_range)
-                    zero = np.zeros((t.shape[0], 1))
                     plot_type = self.plot_func[variable][i].type
+                    if plot_type == PlotType.POINT:
+                        if self.t_to_plot == self.t[i]:
+                            t = self.t[i]
+                    else:
+                        t = self.t[i]
```

Comment:
It seems that there is an else missing here. What happens to t if plot_type is POINT but not equal to self.t? `t` seems to be left undefined

Reply:
At the very least, define it to `None` to specify that we understand that `t` is voluntarily left undefined
`t = self.t[i] if self.t_to_plot == self.t[i] else None`